File: C++/Condition.cpp
#include <iostream>
#include "Header.hpp"

using namespace std;

void Condition(){
	bool test1 = true;
	bool test2 = false;

	cout <<
	endl <<
	"これから条件分岐を試します" << endl <<
	endl;

	if (test1) {
		cout << "どうやらtest1は真のようです" << endl;
	}

	if (!test1) {
		cout << "どうやらtest1は偽のようです" << endl;
	}
	else if (test2) {
		cout << "どうやらtest2は真のようです" << endl;
	}
	else if (!test2 && test1) {
		cout << "真偽が混在しているようです" << endl;
	}
	else {
		cout << "どれでもないようです" << endl;
	}


	int val = 8;
	switch (val) {
		case 0:
		case 1:
			cout << to_string(val)+"は特別な値である" << endl;
			break;
		case 8:
		case 9:
		case 10:
			cout << "これに隣接する数にも言えたりするが,";
			// breakを省くと,case 8,9,10を実行すれば,直下のcase 4,6も実行する
		case 4:
		case 6:
			cout << to_string(val)+"は素数ではない" << endl;
			break;
		case 2:
		case 3:
		case 5:
		case 7:
			cout << to_string(val)+"は素数である" << endl;
			break;
	}

	cout << endl << endl;

}

File: Fortran/Condition.f90
subroutine Condition

 logical::test1 = .true.
 logical::test2 = .false.
 integer val

 print *
 print '("これから条件分岐を試します")'

 if (test1) then
  print '("どうやらtest1は真のようです")'
 end if

 if (.not.test1) then
  print '("どうやらtest1は偽のようです")'
 else if (test2) then
  print '("どうやらtest2は真のようです")'
 else if (.not.test2 .and. test1) then
  print '("真偽が混在しているようです")'
 else
  print '("どれでもないようです")'
 end if

 val = 8
 ! valはinteger,character,logicalの何れかでなければならない
 select case (val)
  case (:-1)
   print '("負の値は受け付けない")'
  case (0:1)
   print '(i2,"は特別な値である")',val
  case (8:10)
   print '(i2,"は連続して素数ではない")',val
  case (4,6)
   print '(i2,"は素数ではない")',val
  case (2,3,5,7)
   print '(i2,"は素数である")',val
  case default
   print '("整数以外は受け付けない")'
 end select

end subroutine Condition